Order of advertising the auth mechanisms (#2745)
esl/MongooseIM: doc/Advanced-configuration.md

    All SCRAM-SHA mechanisms support channel binding and are advertised as a separate authentication mechanisms that is suffixed with `-PLUS`.
    Please note that the list of advertised authentication mechanisms is filtered out by the supported password formats to assure that it is possible to authenticate using authentication mechanisms that are offered.
    * **Warning:** This list is still filtered by [auth backends capabilities](#authentication-backend-capabilities)

>     * **Valid values:** `cyrsasl_scram_sha512_plus, cyrsasl_scram_sha512, cyrsasl_scram_sh384_plus, cyrsasl_scram_sh384,cyrsasl_scram_sha256_plus, cyrsasl_scram_sha256, cyrsasl_scram_sha224_plus, cyrsasl_scram_sha224, cyrsasl_scram_sha1_plus, cyrsasl_scram_sha1, cyrsasl_plain, cyrsasl_anonymous, cyrsasl_oauth, cyrsasl_external, cyrsasl_digest`

Reviewer: ```suggestion
    * **Valid values:** `cyrsasl_scram_sha512_plus, cyrsasl_scram_sha512, cyrsasl_scram_sha384_plus, cyrsasl_scram_sha384, cyrsasl_scram_sha256_plus, cyrsasl_scram_sha256, cyrsasl_scram_sha224_plus, cyrsasl_scram_sha224, cyrsasl_scram_sha1_plus, cyrsasl_scram_sha1, cyrsasl_plain, cyrsasl_anonymous, cyrsasl_oauth, cyrsasl_external, cyrsasl_digest`
```

>     * **Default:** `[cyrsasl_scram_sha512_plus, cyrsasl_scram_sha512, cyrsasl_scram_sh384_plus, cyrsasl_scram_sh384,cyrsasl_scram_sha256_plus, cyrsasl_scram_sha256, cyrsasl_scram_sha224_plus, cyrsasl_scram_sha224, cyrsasl_scram_sha1_plus, cyrsasl_scram_sha1, cyrsasl_plain, cyrsasl_anonymous, cyrsasl_oauth]`

Reviewer: ```suggestion
    * **Default:** `[cyrsasl_scram_sha512_plus, cyrsasl_scram_sha512, cyrsasl_scram_sha384_plus, cyrsasl_scram_sha384, cyrsasl_scram_sha256_plus, cyrsasl_scram_sha256, cyrsasl_scram_sha224_plus, cyrsasl_scram_sha224, cyrsasl_scram_sha1_plus, cyrsasl_scram_sha1, cyrsasl_plain, cyrsasl_anonymous, cyrsasl_oauth]`
```

>         Please note that configuring the `sasl_mechanisms` parameter will take the precedence over the default list.

Reviewer: ```suggestion
        Please note that configuring the `sasl_mechanisms` parameter will take precedence over the default list.
```

>         Should more than one parameter be configured in the list of `sasl_mechanisms`, the order of how they are listed in the config will be taken as the order of advertising the mechanisms.

Reviewer: ```suggestion
        Should more than one parameter be configured in the list of `sasl_mechanisms`, the order of how they are listed in the config will be taken as the order in which they are advertised.
```
